content/src/service/synchronization/ClusterUtils.ts

```diff
@@ -1,38 +1,26 @@
-import ms from "ms"
-import log4js from "log4js"
+import { retry } from "@katalyst/content/helpers/FetchHelper";
 import { ContentServerClient } from "./clients/contentserver/ContentServerClient";
 import { ContentCluster } from "./ContentCluster";
-import { delay } from "decentraland-katalyst-commons/src/util";
-
-
-const LOGGER = log4js.getLogger('ClusterUtils');
 
 /**
  * This method tries to execute a request on all cluster servers, until one responds successfully
  */
-export async function tryOnCluster<T>(execution: (server: ContentServerClient) => Promise<T>, cluster: ContentCluster, options?: { retries?: number, preferred?: ContentServerClient}): Promise<T> {
+export async function tryOnCluster<T>(execution: (server: ContentServerClient) => Promise<T>, cluster: ContentCluster, description: string, options?: { retries?: number, preferred?: ContentServerClient}): Promise<T> {
     // Re order server list
     const servers = reorderAccordingToPreference(cluster.getAllActiveServersInCluster(), options?.preferred);
 
     // Calculate amount of retries. Default is one
     let retries = options?.retries ?? 1
 
-    while (retries >= 0) {
+    return retry(async () => {
         // Try on every cluster server, until one answers the request
         for (const server of servers) {
             try {
                 return await execution(server)
             } catch (error) { }
         }
-        // Wait a little before retrying
-        retries--;
-        if (retries >= 0) {
-            await delay(ms("1s"))
-            LOGGER.info("All calls to other servers failed. Going to retry.")
-        }
-    }
-
-    throw new Error(`Tried to execute request on all servers on the cluster, but they all failed`)
+        throw new Error(`Tried to ${description} on all servers on the cluster, but they all failed`)
```

Comment:
Same here, we are not informing what's the error's cause.

Reply:
We are now logging the error. Since we are trying on many servers, I'm not sure we should so more than logging.